packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { store as blocksStore } from '@wordpress/blocks';
+import {
+	PanelBody,
+	__experimentalHStack as HStack,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../lock-unlock';
+import InspectorControls from '../components/inspector-controls';
+
+export const BlockBindingsPanel = ( { metadata } ) => {
+	const { bindings } = metadata || {};
+	const { sources } = useSelect( ( select ) => {
+		const _sources = unlock(
+			select( blocksStore )
+		).getAllBlockBindingsSources();
+
+		return {
+			sources: _sources,
+		};
+	}, [] );
+
+	if ( ! bindings ) {
+		return null;
+	}
+
+	return (
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Bindings' ) }
+				className="components-panel__block-bindings-panel"
+			>
+				<ItemGroup isBordered isSeparated size="large">
+					{ Object.keys( bindings ).map( ( key ) => {
+						return (
+							<Item key={ key }>
+								<HStack>
+									<span>{ key }</span>
+									<span className="components-item__block-bindings-source">
+										{ sources[ bindings[ key ].source ] ? (
+											sources[ bindings[ key ].source ]
+												.label
+										) : (
+											<span className="error">
+												Error: Unknown Source
```

Comment:
This should be translatable.

What about the case when the source gets registered only on the server? It's all custom sources these days because the client-side API is still private. @SantosGuillamot and I were discussing exposing the registered sources from the server in a different PR

Reply:
The sources we are consuming in the editor need to be registered in JS as well. They provide functions to know how to get the values and how to update them in the editor. One example could be the [post-meta source](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/bindings/post-meta.js) that is registered [here](https://github.com/WordPress/gutenberg/blob/4829a87a65b171ff563fd31bb6aa490ce8ab8069/packages/editor/src/bindings/index.js#L14-L15).

Registering in the client is not mandatory, and it doesn't mean it won't work on the server. So it is not exactly an error. I believe right now we are using "Dynamic Data" as a generic message, but we could also fallback to the key maybe.

In the future, as Greg mentions, we would like to reuse the server registration somehow, but that will be handled in a separate PR.

Reply:
Ok got it! Thanks so much for the explanation, that's very helpful 😄
I can revise to use the name in the binding when a label isn't present 👍